File: 736_partition_labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string t) {
        int u[128] = {}, j = 0, d = 0;
        for (int i = 0; i < t.size(); i++) u[t[i]] = i;
        vector<int> r;
        for (int i = 0; i < t.size(); i++) {
            j = max(j, u[t[i]]);
            if (i == j) r.push_back(i - d + 1), d = i + 1;
        }
        return r;
    }
};

File: 736_partition_labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        int[] f = new int[128];
        for (int i = 0; i < s.length(); i++) f[s.charAt(i)] = i;
        List<Integer> r = new ArrayList<>();
        for (int i = 0, e = 0, b = 0; i < s.length(); i++) {
            e = Math.max(e, f[s.charAt(i)]);
            if (i == e) {
                r.add(i - b + 1);
                b = i + 1;
            }
        }
        return r;
    }
}
